internal/checker/terraform/rules/aws/verify_aws_dms_endpoint_have_ssl_configured.rego

```diff
@@ -0,0 +1,77 @@
+# METADATA
+# title: "Verify AWS Database Migration Service endpoints have SSL configured"
+# description: "This policy ensures the secure and encrypted communication of Amazon Web Services (AWS) Database Migration Service (DMS) endpoints by verifying the proper configuration of SSL (Secure Sockets Layer)."
+# scope: package
+# related_resources:
+# - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_endpoint.html
+# custom:
+#   id: CB_TFAWS_364
+#   severity: HIGH
+package lib.terraform.CB_TFAWS_364
+
+isvalid(block) {
+	block.Type == "resource"
+	block.Labels[_] == "aws_dms_endpoint"
+}
+
+resource[resource] {
+	block := pass[_]
+	resource := concat(".", block.Labels)
+}
+
+resource[resource] {
+	block := fail[_]
+	resource := concat(".", block.Labels)
+}
+
+isEngineValidForTypeSource {
+	resource := input[_]
+	resource.Type == "resource"
+	validEngine := ["s3", "azuredb"]
```

Comment:
Kindly check if lines 30-31 and 38-39 are required (Perform a scan using checkov and see if it needs to check the value of engine_name).

If the value of the engine_name does not affect the pass or fail result, the lines may be removed.